src/snowflake/connector/constants.py

```diff
@@ -17,25 +26,58 @@
 class FieldType(NamedTuple):
     name: str
     dbapi_type: list[int]
-
-
-FIELD_TYPES: list[FieldType] = [
-    FieldType(name="FIXED", dbapi_type=[DBAPI_TYPE_NUMBER]),
-    FieldType(name="REAL", dbapi_type=[DBAPI_TYPE_NUMBER]),
-    FieldType(name="TEXT", dbapi_type=[DBAPI_TYPE_STRING]),
-    FieldType(name="DATE", dbapi_type=[DBAPI_TYPE_TIMESTAMP]),
-    FieldType(name="TIMESTAMP", dbapi_type=[DBAPI_TYPE_TIMESTAMP]),
-    FieldType(name="VARIANT", dbapi_type=[DBAPI_TYPE_BINARY]),
-    FieldType(name="TIMESTAMP_LTZ", dbapi_type=[DBAPI_TYPE_TIMESTAMP]),
-    FieldType(name="TIMESTAMP_TZ", dbapi_type=[DBAPI_TYPE_TIMESTAMP]),
-    FieldType(name="TIMESTAMP_NTZ", dbapi_type=[DBAPI_TYPE_TIMESTAMP]),
-    FieldType(name="OBJECT", dbapi_type=[DBAPI_TYPE_BINARY]),
-    FieldType(name="ARRAY", dbapi_type=[DBAPI_TYPE_BINARY]),
-    FieldType(name="BINARY", dbapi_type=[DBAPI_TYPE_BINARY]),
-    FieldType(name="TIME", dbapi_type=[DBAPI_TYPE_TIMESTAMP]),
-    FieldType(name="BOOLEAN", dbapi_type=[]),
-    FieldType(name="GEOGRAPHY", dbapi_type=[DBAPI_TYPE_STRING]),
-]
+    pa_type: Callable[[], DataType]
```

Comment:
Same question. Why do we need that lambda with no parameter?

Reply:
Since `pyarrow` is an optional dependency we cannot resolve `pyarrow.int64` for sure.
If we didn't have `pyarrow` installed then the lookup invoked by `pyarrow.int64` would trigger an error, so I wrap these definitions into lambdas. This way I still get [lookup magic from snowflake.connector.options](https://github.com/snowflakedb/snowflake-connector-python/blob/b132ab1eb8a1b193191c1e7f0897cb7df618dedc/src/snowflake/connector/options.py#L38) with nice errors if they are needed, but I don't resolve them until I necessarily have to